mypy/checkexpr.py

```diff
@@ -966,10 +966,7 @@ def visit_op_expr(self, e: OpExpr) -> Type:
             # Expressions of form [...] * e get special type inference.
             return self.check_list_multiply(e)
         if e.op == '%':
-            if isinstance(e.left, BytesExpr):
-                return self.strfrm_checker.check_byte_interpolation(e.left, e.right)
-            elif isinstance(e.left, StrExpr):
-                return self.strfrm_checker.check_str_interpolation(cast(StrExpr, e.left), e.right)
+            return self.strfrm_checker.check_str_interpolation(e.left, e.right)
```

Comment:
I added the isinstance check back. It was my oversite that non StrExpr-like objects would get through to this point.